csrc/multidevice/ipc_handle.cpp

```diff
@@ -0,0 +1,150 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <cuda_utils.h>
+#include <multidevice/communicator.h>
+#include <multidevice/ipc_handle.h>
+
+namespace nvfuser {
+
+namespace {
+
+template <typename T>
+std::vector<uint8_t> toBytes(const T& data) {
+  return std::vector<uint8_t>(
+      reinterpret_cast<const uint8_t*>(&data),
+      reinterpret_cast<const uint8_t*>(&data) + sizeof(T));
+}
+
+template <typename T>
+const T& fromBytes(const std::vector<uint8_t>& bytes) {
+  return *reinterpret_cast<const T*>(bytes.data());
+}
+
+} // namespace
+
+IpcHandle::IpcHandle(at::Tensor tensor)
+    : ptr_(tensor.data_ptr()), rank_(Communicator::getInstance().deviceId()) {
+  size_t alloc_length = 0;
```

Comment:
By length, you mean the number of elements or size in bytes? (Thus, a soft request to rename)

Reply:
renamed to `psize` as in the doc https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g64fee5711274a2a0573a789c94d8299b